mlflow/pmdarima.py

```diff
@@ -0,0 +1,241 @@
+            )
+            default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+        else:
+            default_reqs = None
+        conda_env, pip_requirements, pip_constraints = _process_pip_requirements(
+            default_reqs, pip_requirements, extra_pip_requirements
+        )
+    else:
+        conda_env, pip_requirements, pip_constraints = _process_conda_env(conda_env)
+
+    with open(os.path.join(path, _CONDA_ENV_FILE_NAME), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    if pip_constraints:
+        write_to(os.path.join(path, _CONSTRAINTS_FILE_NAME), "\n".join(pip_constraints))
+
+    write_to(os.path.join(path, _REQUIREMENTS_FILE_NAME), "\n".join(pip_requirements))
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def log_model(
+    pmdarima_model,
+    artifact_path,
+    conda_env=None,
+    registered_model_name=None,
+    signature: ModelSignature = None,
+    input_example: ModelInputExample = None,
+    await_registration_for=DEFAULT_AWAIT_MAX_SLEEP_SECONDS,
+    pip_requirements=None,
+    extra_pip_requirements=None,
+    **kwargs,
+):
+    """
+    :param pmdarima_model:
+    :param artifact_path:
+    :param conda_env:
+    :param registered_model_name:
+    :param signature:
+    :param input_example:
+    :param await_registration_for:
+    :param pip_requirements:
+    :param extra_pip_requirements:
+    :param kwargs:
+
+    :return:
+    """
+
+    return Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.pmdarima,
+        registered_model_name=registered_model_name,
+        pmdarima_model=pmdarima_model,
+        conda_env=conda_env,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        **kwargs,
+    )
+
+
+def load_model(model_uri, dst_path=None):
+    """
+    :param model_uri:
+    :param dst_path:
+    """
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    flavor_conf = _get_flavor_configuration(model_path=local_model_path, flavor_name=FLAVOR_NAME)
+    pmdarima_model_file_path = os.path.join(
+        local_model_path, flavor_conf.get(_MODEL_BINARY_KEY, _MODEL_BINARY_FILE_NAME)
+    )
+
+    return _load_model(pmdarima_model_file_path)
+
+
+def _save_model(model, path):
+
+    with open(path, "wb") as f:
+        pickle.dump(model, f)
+
+
+def _load_model(path):
+
+    with open(path, "rb") as pickled_model:
+        model = pickle.load(pickled_model)
+    return model
+
+
+def _load_pyfunc(path):
+
+    return _PmdarimaModelWrapper(_load_model(path))
+
+
+class _PmdarimaModelWrapper:
+    def __init__(self, pmdarima_model):
+        self.pmdarima_model = pmdarima_model
+
+    def predict(self, dataframe):
+
+        attrs = dataframe.to_dict(orient="index").get(0)
```

Comment:
Do we want to verify that the dataframe contains exactly one row?

Reply:
definitely. also added a schema validation check and added a test to validate both.